test/integration/ext_deletion_test.go

```diff
@@ -0,0 +1,100 @@
+// +build integration,etcd
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	newproject "github.com/openshift/origin/pkg/cmd/admin/project"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	expapi "k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+const projName = "ext-proj"
+
+func init() {
+	testutil.RequireEtcd()
+}
+
+func TestExtDeletion(t *testing.T) {
+	_, clusterAdminKubeConfig, err := testserver.StartTestMaster()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	clusterAdminClient, err := testutil.GetClusterAdminClient(clusterAdminKubeConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	clusterAdminKubeClient, err := testutil.GetClusterAdminKubeClient(clusterAdminKubeConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// create the containing project
+	newProjectOptions := &newproject.NewProjectOptions{
+		Client:      clusterAdminClient,
+		ProjectName: projName,
+		AdminRole:   bootstrappolicy.AdminRoleName,
+		AdminUser:   "admin",
+	}
+	if err := newProjectOptions.Run(false); err != nil {
+		t.Fatalf("unexpected error creating the project: %v", err)
+	}
+
+	extensionsClient := clusterAdminKubeClient.Extensions()
+
+	// create the extensions resources
```

Comment:
while you're at it, create the HPA and Job using the project admin client returned from `CreateNewProject`, to make sure default policy allows project admins to create those things